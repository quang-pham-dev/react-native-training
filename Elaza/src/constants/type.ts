export const SIZE = {
  TINY: 'tiny',
  SMALL: 'small',
  MEDIUM: 'medium',
  LARGE: 'large',
  BASE: 'base',
}

export const HEADING_TYPE = {
  H1: 'h1',
  H2: 'h2',
  H3: 'h3',
  H4: 'h4',
}

export const PARAGRAPH_TYPE = {
  TINY: 'tiny',
  SMALL: 'small',
  BASE: 'base',
  MEDIUM: 'medium',
  LARGER: 'larger',
}

export const BUTTON_TYPE = {
  CONFIRM: 'confirm',
  CANCEL: 'cancel',
  LINK: 'link',
  SUBMIT: 'submit',
}
