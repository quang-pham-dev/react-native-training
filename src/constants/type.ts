export const SIZE = {
  TINY: 'tiny',
  SMALL: 'small',
  MEDIUM: 'medium',
  LARGE: 'large',
  BASE: 'base',
}

export const HEADING_TYPE = {
  H1: 'h1',
  H2: 'h2',
  H3: 'h3',
  H4: 'h4',
}

export const PARAGRAPH_TYPE = {
  TINY: 'tiny',
  SMALL: 'small',
  BASE: 'base',
  MEDIUM: 'medium',
  LARGER: 'larger',
  BRAND: 'brand',
  PRODUCT_NAME_AND_TYPE: 'productNameAndType',
  PRODUCT_PRICE: 'productPrice',
  ALREADY_ACCOUNT: 'alreadyAccount',
  ALREADY_SIGN_IN: 'alreadySignIn',
  CONFIRM_AGREE: 'confirmAgree',
  TERM_AND_CONDITION: 'termAndCondition',
  PROFILE_USERNAME: 'profileUsername',
  PROFILE_VERIFIED: 'profileVerified',
  PROFILE_ITEM: 'profileItem',
  BRAND_DETAIL_TITLE: 'brandDetailTitle',
  BRAND_DETAIL_ITEM_COUNT: 'brandDetailItemCount',
}

export const BUTTON_TYPE = {
  BOTTOM: 'bottom',
  SOCIAL: 'social',
}
