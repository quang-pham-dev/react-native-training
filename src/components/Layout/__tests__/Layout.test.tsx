import React from 'react'
import {render} from '@testing-library/react-native'

import LayoutStyled from '../Layout.styles'
import HeadingStyled from '@components/Heading/Heading.styles'

import {HEADING_TYPE} from '@constants/type'

describe('Layout', () => {
  test('should render Main correctly correctly', () => {
    const component = render(
      <LayoutStyled.Main>
        <HeadingStyled type={HEADING_TYPE.H1}>
          Lorem ipsum dolor sit a
        </HeadingStyled>
      </LayoutStyled.Main>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('should render Base correctly correctly', () => {
    const component = render(
      <LayoutStyled.Base>
        <HeadingStyled type={HEADING_TYPE.H1}>
          Lorem ipsum dolor sit a
        </HeadingStyled>
      </LayoutStyled.Base>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('should render PaddingBottom35 correctly correctly', () => {
    const component = render(
      <LayoutStyled.PaddingBottom35>
        <HeadingStyled type={HEADING_TYPE.H1}>
          Lorem ipsum dolor sit a
        </HeadingStyled>
      </LayoutStyled.PaddingBottom35>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('should render PaddingBottom38 correctly correctly', () => {
    const component = render(
      <LayoutStyled.PaddingBottom38>
        <HeadingStyled type={HEADING_TYPE.H1}>
          Lorem ipsum dolor sit a
        </HeadingStyled>
      </LayoutStyled.PaddingBottom38>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
