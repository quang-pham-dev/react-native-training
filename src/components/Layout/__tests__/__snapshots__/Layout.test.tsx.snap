// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render Base correctly correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F5F6FA",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingHorizontal": 20,
        "width": "100%",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#1D1E20",
          "fontFamily": "Inter-SemiBold",
          "fontSize": 28,
          "lineHeight": 30,
          "textAlign": "center",
        },
      ]
    }
    type="h1"
  >
    Lorem ipsum dolor sit a
  </Text>
</View>
`;

exports[`Layout should render Main correctly correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingHorizontal": 20,
        "width": "100%",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#1D1E20",
          "fontFamily": "Inter-SemiBold",
          "fontSize": 28,
          "lineHeight": 30,
          "textAlign": "center",
        },
      ]
    }
    type="h1"
  >
    Lorem ipsum dolor sit a
  </Text>
</View>
`;

exports[`Layout should render PaddingBottom35 correctly correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F5F6FA",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 35,
        "paddingHorizontal": 20,
        "width": "100%",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#1D1E20",
          "fontFamily": "Inter-SemiBold",
          "fontSize": 28,
          "lineHeight": 30,
          "textAlign": "center",
        },
      ]
    }
    type="h1"
  >
    Lorem ipsum dolor sit a
  </Text>
</View>
`;

exports[`Layout should render PaddingBottom38 correctly correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F5F6FA",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 38,
        "paddingHorizontal": 20,
        "width": "100%",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#1D1E20",
          "fontFamily": "Inter-SemiBold",
          "fontSize": 28,
          "lineHeight": 30,
          "textAlign": "center",
        },
      ]
    }
    type="h1"
  >
    Lorem ipsum dolor sit a
  </Text>
</View>
`;
